### PR TITLE
stdio: fix fgets(), add gets()

### LINKED ISSUE #1489: `fgets()` behavior and implementation diverges from ISO C standard
According to ISO C standard:
https://pubs.opengroup.org/onlinepubs/009696699/functions/fgets.html
The fgets() function shall read bytes from stream into the array pointed to by s until n-1 bytes are read, or a <newline> is read and transferred to s, or an end-of-file condition is encountered. A null byte shall be written immediately after the last byte read into the array. If the end-of-file condition is encountered before any bytes are read, the contents of the array pointed to by s shall not be changed.

Meanwhile, in libphoenix `fgets()` behavior is divergent for n=1. The function reads single byte, which is outcome of implementation https://github.com/phoenix-rtos/libphoenix/blob/3f0d10611013ed992fb351d0e039a8e6655aa647/stdio/file.c#L736

Functionality was tested with the following code:
```c
#include <stdio.h>

int main(void)
{
	char buff[2];

	fgets(buff, 1, stdin);
	printf("output: %u\t%u\n", buff[0], buff[1]);

	return 0;
}
```

### PR DESCRIPTION
**Pull request by julianuziemblo**
Fixes: phoenix-rtos/phoenix-rtos-project#1489

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: `ia32-generic-qemu`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes: https://github.com/phoenix-rtos/phoenix-rtos-tests/pull/479
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: stdio/file.c [hunk 1]
```diff
@@ -732,18 +732,26 @@ int fputc(int c, FILE *stream)
 }
 
 
-char *fgets_unlocked(char *str, int n, FILE *stream)
+/* if n < 1 -> no limit */
+static char *_fgets(char *str, int n, FILE *stream)
 {
-	int c, i = 0;
+	int c, i = -1;
```
Comment (nalajcie) [inline stdio/file.c]: I think we don't need to start from `-1` here.

Also: for n == 1 this function behaves incorrectly - consumes byte from the string when it shouldn't (it was previously also probably incorrect, still - you're meddling with `n` and `i` so you should spot and fix it).

Edit: you've explicitly stated that this PR fixes the `n=1` case which is not the case.

I'm against merging this until tests for corner cases are added. Start by adding tests to verify the problem exists, change PR to draft until the issues are fixed.

**Review by nalajcie — CHANGES_REQUESTED** [review]
Adding multiple people before even the test were run (and failed)? Please be more mindful to the potential reviewers,

File: stdio/file.c [hunk 2]
```diff
@@ -734,21 +734,36 @@ int fputc(int c, FILE *stream)
 
 char *fgets_unlocked(char *str, int n, FILE *stream)
 {
+	if (n < 1) {
+		return NULL;
+	}
+
+	if (n == 1) {
```
Comment (etiaro) [inline stdio/file.c]: I don't think this corner case requires separate handling here

**Review by etiaro — COMMENTED** [review]
Should be split into 2 separate, atomic commits. Other than that, lgtm.

**Review by etiaro — APPROVED** [review]
